### PR TITLE
Fix transparency over heightmap

### PR DESCRIPTION
**Pull request by iche033**
# 🦟 Bug fix

Fixes https://github.com/gazebosim/gz-rendering/issues/789

## Summary

Semi-transparent objects were always rendered behind the heightmap as seen in #789. This PR fixes the issue by setting a new render queue for objects with transparency as recommended in https://forums.ogre3d.org/viewtopic.php?t=96950

To test with gazebo:

1. launch gazebo (fortress) with the heightmap world:

```sh
ign gazebo -v 4 heightmap.sdf
```

1. Select Translate tool and click on any of the models. You should see the RGB axes rendered properly over the heightmap.

1. Right click on the model -> View > Transparent. The model should also be rendered properly over heightmap

transparency_over_heightmap

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ogre2/src/Ogre2Mesh.cc [hunk 1]
```diff
@@ -355,8 +355,24 @@ void Ogre2SubMesh::SetMaterialImpl(MaterialPtr _material)
   // Pbs Hlms material
   else
   {
-    this->ogreSubItem->setDatablock(
-        static_cast<Ogre::HlmsPbsDatablock *>(derived->Datablock()));
+    auto datablock =
```
Comment (ahcorde) [inline ogre2/src/Ogre2Mesh.cc:358]: check datablock ?

File: test/integration/heightmap.cc [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    bool result = common::env("MESA_GL_VERSION_OVERRIDE", value, true);
+    if (result && value == "3.3")
+    {
+      igndbg << "Test is run on machine with software rendering or mesa driver "
+             << "Skipping test. " << std::endl;
+      return;
+    }
+#endif
+
+  // Test: the scene consists of red background, semi-transprent green box
```
Comment (ahcorde) [inline test/integration/heightmap.cc:73]:
```suggestion
  // Test: the scene consists of red background, semi-transparent green box
```

File: test/integration/heightmap.cc [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  // check the rgb value of the image at position of the box
+  // it should be blending of green box with blue heightmap and no
+  // red background
+
+  // create and populate scene
+  RenderEngine *engine = rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
```
Comment (ahcorde) [inline test/integration/heightmap.cc:84]:
```suggestion
    igndbg << "Engine '" << _renderEngine
           << "' is not supported" << std::endl;
```

File: test/integration/heightmap.cc [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  root->AddChild(camera);
+
+  // create ImageHeightmap
+  auto data = std::make_shared<common::ImageHeightmap>();
+  data->Load(common::joinPaths(TEST_MEDIA_PATH, "heightmap_bowl.png"));
+
+  HeightmapDescriptor desc;
+  desc.SetName("example_bowl");
+  desc.SetData(data);
+  desc.SetSize({ 17, 17, 7.0 });
```
Comment (ahcorde) [inline test/integration/heightmap.cc:111]:
```suggestion
  desc.SetSize({17, 17, 7.0});
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: test/integration/heightmap.cc (hunk 4 above)
Comment (iche033) [reply test/integration/heightmap.cc]: [f33d5c2](https://github.com/gazebosim/gz-rendering/pull/811/commits/f33d5c21898b95303c9fd7e2ab21fd2091d4a88d)

File: test/integration/heightmap.cc (hunk 3 above)
Comment (iche033) [reply test/integration/heightmap.cc]: [f33d5c2](https://github.com/gazebosim/gz-rendering/pull/811/commits/f33d5c21898b95303c9fd7e2ab21fd2091d4a88d)

File: test/integration/heightmap.cc (hunk 2 above)
Comment (iche033) [reply test/integration/heightmap.cc]: fixed. [f33d5c2](https://github.com/gazebosim/gz-rendering/pull/811/commits/f33d5c21898b95303c9fd7e2ab21fd2091d4a88d)

File: ogre2/src/Ogre2Mesh.cc (hunk 1 above)
Comment (iche033) [reply ogre2/src/Ogre2Mesh.cc:358]: yep, added check. [f33d5c2](https://github.com/gazebosim/gz-rendering/pull/811/commits/f33d5c21898b95303c9fd7e2ab21fd2091d4a88d)

File: ogre2/src/Ogre2Mesh.cc [hunk 5]
```diff
@@ -355,8 +355,27 @@ void Ogre2SubMesh::SetMaterialImpl(MaterialPtr _material)
...
+        static_cast<Ogre::HlmsPbsDatablock *>(derived->Datablock());
+    if (databock)
```
Comment (ahcorde) [inline ogre2/src/Ogre2Mesh.cc:360]:
```suggestion
    if (datablock)
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: ogre2/src/Ogre2Mesh.cc (hunk 5 above)
Comment (iche033) [reply ogre2/src/Ogre2Mesh.cc]: oops fixed. [3312c33](https://github.com/gazebosim/gz-rendering/pull/811/commits/3312c335cd7856d09b46dcdf9772d56747ceb140)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**